### PR TITLE
feat(cra): improve TypeScript docgen support

### PR DESCRIPTION
**Pull request by mrmckeb**

### CODE REVIEW

File: packages/preset-create-react-app/.storybook/presets.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+const path = require('path');
```
Comment (shilman) [inline packages/preset-create-react-app/.storybook/presets.js]: why?

File: packages/preset-create-react-app/README.md [hunk 2]
```diff
@@ -26,15 +26,35 @@ In most cases, this preset will find your `react-scripts` package, even if it's
...
+
+If set to `{}`, it will be enabled with default Create React App settings.
+
+```js
+module.exports = [
+  {
+    name: '@storybook/preset-create-react-app',
+    options: {
+      scriptsPackageName: '@my/react-scripts',
+      tsDocgenLoaderOptions: true,
```
Comment (shilman) [inline packages/preset-create-react-app/README.md]: `true`? or `{}`? i'm confused

File: packages/preset-create-react-app/.storybook/presets.js (hunk 1 above)
Comment (mrmckeb) [reply packages/preset-create-react-app/.storybook/presets.js]: That was from testing, removed.

File: packages/preset-create-react-app/README.md (hunk 2 above)
Comment (mrmckeb) [reply packages/preset-create-react-app/README.md]: Sorry, that was me playing around and I forgot to revert. Fixed.

**Pull request merged**